File: chapter02/sanghyo/MovieReservationProgram/MovieReservationProgram/Model/Reservation.swift
//
//  Reservation.swift
//  MovieReservationProgram
//
//  Created by sanghyo on 11/29/23.
//

import Foundation

struct Reservation {
    private let customer: Customer
    private let screening: Screening
    private let fee: Money
    private let audienceCount: Int
    
    init(customer: Customer, screening: Screening, fee: Money, audienceCount: Int) {
        self.customer = customer
        self.screening = screening
        self.fee = fee
        self.audienceCount = audienceCount
    }
}

File: chapter02/hm4725/src/Reservation.cc
#include "Reservation.hh"

Reservation::Reservation(std::shared_ptr<Customer> customer,
                         std::shared_ptr<Screening> screening,
                         std::shared_ptr<Money> fee, int audienceCount)
    : customer(customer),
      screening(screening),
      fee(fee),
      audienceCount(audienceCount) {}